src/indexer/services/substrateIndexer/substrateIndexer.ts

```diff
@@ -46,12 +66,66 @@ export class SubstrateIndexer {
           toBlock = fromBlock + this.pastEventsQueryInterval
         }
 
-        await this.saveDataToDb(this.domain.id, toBlock.toString())
+        const blockHash = await this.provider.rpc.chain.getBlockHash(fromBlock)
+        const signedBlock = await this.provider.rpc.chain.getBlock(blockHash)
+        const at = await this.provider.at(blockHash)
+        const timestamp = Number((await at.query.timestamp.now()).toString())
+        const allRecords = await at.query.system.events()
+
+        // we get the proposal execution events - ts-ignore because of allRecords
+        // @ts-ignore
+        const proposalExecutionEvents = getSubstrateEvents(SygmaPalleteEvents.ProposalExecution, allRecords) as Array<ProposalExecutionEvent>
+
+        // we get the deposit events - ts-ignore because of allRecords
+        // @ts-ignore
+        const depositEvents = getSubstrateEvents(SygmaPalleteEvents.Deposit, allRecords) as Array<DepositEvent>
+
+        // we get the index of the section in the extrinsic
+        const sectionIndex = signedBlock.block.extrinsics.findIndex(ex => ex.method.section === "sygmaBridge")
+
+        // this is our identifier for the tx
+        const txIdentifier = `${fromBlock}-${sectionIndex}` //this is like the txHash but for the substrate
+
+        if (proposalExecutionEvents.length) {
```

Comment:
You don't need this check.
If proposalExecutionEvents is [] it will just skip anyway.

Reply:
indeed, changed